### PR TITLE
Provide more helpful error when using unsupported api version

### PR DESCRIPTION
**Pull request by feihujiang**
#10318

### CODE REVIEW

**Comment by feihujiang** [comment]
@satnam6502 @mikedanese Could you please review this PR?

**Comment by satnam6502** [comment]
Other than perhaps saying "API version" instead of "APIVersion" (no need to make this change) LGTM.

**Comment by erictune** [comment]
OK to test

**Comment by mikedanese** [comment]
LGTM. @k8s-bot ok to test

**Comment by satnam6502** [comment]
I think we now need two TLs. @thockin @dchen1107

**Comment by satnam6502** [comment]
Risk: low.

File: pkg/kubectl/cmd/util/factory.go [hunk 1]
```diff
@@ -267,6 +268,9 @@ func (c *clientSwaggerSchema) ValidateBytes(data []byte) error {
 	if err != nil {
 		return err
 	}
+	if ok := registered.IsRegisteredAPIVersion(version); !ok {
+		return fmt.Errorf("APIVersion %q isn't supported, only supports APIVersions %q", version, registered.RegisteredVersions)
```
Comment (thockin) [inline pkg/kubectl/cmd/util/factory.go]: Humanize to "API version %q ..."

**Comment by feihujiang** [comment]
@thockin I changed according to your advice, please review this PR again.

**Comment by satnam6502** [comment]
@k8s-bot ok to test

**Comment by thockin** [comment]
LGTM

@lavalamp

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
Shippable is the "F0629 07:51:01.463390 33737 integration.go:871] FAILED: pod never started running timed out waiting for the condition" flake.

**Pull request merged**